### DISCUSSION TITLE
FileRoutes: support for custom directory

### DISCUSSION THREAD

**Discussion by indeyets** [solidjs/solid-start#536] [Ideas]
Being able to customize routes directory of `<FileRoutes>` component would allow to have more flexibility and ability to have several independent routers.

I'm thinking about something like this:

```jsx
<div id="menu">
  <Routes>
    <FileRoutes directory="src/menu-routes" />
  </Routes>
</div>

<div id="body">
  <Routes>
    <FileRoutes directory="src/routes" />
  </Routes>
</div>
```

**Comment by orenelbaum** [comment]
seems to me like a good idea

**Comment by russelgal** [comment]
or dynamic directories to support multiple domains
```js
<div id="body">
  <Routes>
    <FileRoutes directory={(host)=>"src/" + host} />
  </Routes>
</div>

**Comment by AFatNiBBa** [comment]
This would be very useful

**Comment by doeixd** [comment]
In the meantime, you could write a higher order component that wraps the file routes, and makes the changes you desire. Or you could hook into vinxi/vite yourself and make a new filesystem router that meets your needs.
